### ISSUE TITLE
NavigationRail should have a iconType attribute so that only the selected or none of the icons are displayed

### ISSUE THREAD

**Issue by stephane-archer** [flutter/flutter#134944]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

NavigationRail doesn't have `iconType` only `labelType`

### Expected results

add iconType to NavigationRail

### Actual results

no iconType to NavigationRail

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by huycozy** [comment]
Labeling this as a feature request for adding `iconType` to `NavigationRail`. Thank you!

**Comment by opxdelwin** [comment]
@stephane-archer @huycozy could you please elaborate a bit, I don't seem to understand the implementation. Thanks!

**Comment by HansMuller** [comment]
Interesting proposal. If we could sort out the design, we'd welcome a PR.

**Comment by stephane-archer** [comment]
@opxdelwin Yes I was thinking of that. some designs make an accent on the label, not the icon
for example in my case I don't use any icon
![lutme_1 (1)](https://github.com/flutter/flutter/assets/2981437/07bdab92-e4e5-4f2a-97a0-0224ef09c891)

**Comment by opxdelwin** [comment]
@stephane-archer what could be possible combinations? 
```
All Icon + All Label,
All Icon + Selected Label,
All Icon + No Label,
Selected Icon + All Label,
Selected Icon + Selected Label (might not be suitable),
Selected Icon + No Label (might not be suitable),
No Icon + All Label,
No Icon + Selected Label (might not be suitable),
No Icon + No Label (might not be suitable),
```
Let me know if I've missed any combination or you'd have any suggestions... I'll mostly create a PR once confirmed.

**Comment by stephane-archer** [comment]
I think you are correct for all combinations,
I don’t know how should be treated all "might not be suitable” case.

**Comment by opxdelwin** [comment]
as long as the developer doesn't change the type dynamically (i.e at runtime), we could just use `asset` to limit combinations, the other way would be to have a placeholder combination(or like a default config, lets say `all + all`) if type changes to restricted one dynamically in runtime

**Comment by opxdelwin** [comment]
Looks like we'll move forward with [these ](https://github.com/flutter/flutter/issues/134944#issuecomment-1731956519)combinations and `assert()` to deny bad combinations
